common/oatbox/cache/MultipleCacheTrait.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ *
+ */
+
+namespace oat\oatbox\cache;
+
+/**
+ * Caches data in a key-value store
+ *
+ * @access public
+ * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
+ * @package generis
+ */
+trait MultipleCacheTrait
+{
+
+    public function deleteMultiple($keys)
+    {
+        $success = true;
+        foreach ($keys as $key) {
+            $success = $this->delete($key) && $success;
+        }
+        return $success;
+    }
+
+    public function getMultiple($keys, $default = null)
+    {
+        $values = [];
+        foreach ($keys as $key) {
+            $values[$key] = $this->get($key, $default);
+        }
+        return $values;
+    }
+
+    public function setMultiple($values, $ttl = null)
+    {
+        $success = true;
+        foreach ($values as $key => $value) {
+            $success = $this->set($key, $value, $ttl) && $success;
```

Comment:
It may confuse developers. In case of failure all already set values have to be deleted.
The same for delete. All already deleted items have to be restored.

Reply:
I could not find any indication in the PSR on how a partial success/failure should be handled

Reply:
In that case this behavior is up to implementation.
I think rolling back in that case corresponds to best practices.

Reply:
Unfortunately I have no idea how to implement this rollback as filesystems are not transactional